source/m2mbase.cpp

```diff
@@ -780,15 +790,17 @@ void M2MBase::free_resources()
         free(_sn_resource->dynamic_resource_params);
     }
 
-    if (_sn_resource->free_on_delete) {
-        free(_sn_resource->name);
-        free(_sn_resource);
+    if(_sn_resource->identifier_int_type == false) {
```

Comment:
This can't work, the "_sn_resource->free_on_delete" is false on structs living in ROM and the freeing will panic. The original if() needs to be left there.